src/routes/sessions/+layout.svelte

```diff
@@ -13,14 +14,20 @@
 	<svelte:fragment slot="list-items">
 		{#if $sessionsStore && $sessionsStore.length > 0}
 			<!-- Using slice() to reverse $sessionsStore without affecting the original array -->
-			{#each $sessionsStore.slice().reverse() as session (session.id)}
+			{#each $sessionsStore.slice().sort((a, b) => {
+				if (!a.updatedAt || !b.updatedAt) return 0;
+				return a.updatedAt < b.updatedAt ? 1 : -1;
+			}) as session}
```

Comment:
Now that `Sessions` and `Knowledge` have an `updatedAt` field we could sort the store every time we update it:
- `saveKnowledge()`
- `saveSession()`

Something like this maybe...
```typescript
export const saveKnowledge = (knowledge: Knowledge): void => {
  // Retrieve the current knowledges
  const currentKnowledges = get(knowledgeStore) || [];

  // Find the index of the knowledge with the same id, if it exists
  const existingIndex = currentKnowledges.findIndex(k => k.id === knowledge.id);

  if (existingIndex !== -1) {
    // Update the existing knowledge
    currentKnowledges[existingIndex] = knowledge;
  } else {
    // Add the new knowledge if it doesn't exist
    currentKnowledges.push(knowledge);
  }

  // Sort the knowledges by updatedAt in descending order (most recent first)
  const sortedKnowledges = currentKnowledges.sort((a, b) => 
    new Date(b.updatedAt).getTime() - new Date(a.updatedAt).getTime()
  );

  // Update the store with the sorted knowledges
  knowledgeStore.set(sortedKnowledges);
};
```

And then we don't need to `.slice().sort()` on `$sessionStore` or `$knowledgeStore` .

Reply:
I know we talked about potentially DRY'n up these store files but for know let's only abstract the sorting logic. We can do the rest of the refactoring later.